File: system-test/java/src/main/java/com/mycompany/myshop/testkit/driver/adapter/myshop/ui/client/pages/NewOrderPage.java
package com.mycompany.myshop.testkit.driver.adapter.myshop.ui.client.pages;

import com.mycompany.myshop.testkit.driver.adapter.shared.client.playwright.PageClient;

import java.util.regex.Pattern;

public class NewOrderPage extends BasePage {
    private static final String SKU_INPUT_SELECTOR = "[aria-label=\"SKU\"]";
    private static final String QUANTITY_INPUT_SELECTOR = "[aria-label=\"Quantity\"]";
    private static final String COUNTRY_INPUT_SELECTOR = "[aria-label=\"Country\"]";
    private static final String COUPON_CODE_INPUT_SELECTOR = "[aria-label=\"Coupon Code\"]";
    private static final String PLACE_ORDER_BUTTONG_SELECTOR = "[aria-label=\"Place Order\"]";
    private static final String ORDER_NUMBER_REGEX = "Success! Order has been created with Order Number ([\\w-]+)";
    private static final int ORDER_NUMBER_MATCHER_GROUP = 1;
    private static final String ORDER_NUMBER_NOT_FOUND_ERROR = "Could not find order number";

    public NewOrderPage(PageClient pageClient) {
        super(pageClient);
    }

    public void inputSku(String sku) {
        pageClient.fill(SKU_INPUT_SELECTOR, sku);
    }

    public void inputQuantity(String quantity) {
        pageClient.fill(QUANTITY_INPUT_SELECTOR, quantity);
    }

    public void inputCountry(String country) {
        pageClient.fill(COUNTRY_INPUT_SELECTOR, country);
    }

    public void inputCouponCode(String couponCode) {
        pageClient.fill(COUPON_CODE_INPUT_SELECTOR, couponCode);
    }

    public void clickPlaceOrder() {
        pageClient.click(PLACE_ORDER_BUTTONG_SELECTOR);
    }

    public static String getOrderNumber(String successMessageText) {

        var pattern = Pattern.compile(ORDER_NUMBER_REGEX);
        var matcher = pattern.matcher(successMessageText);

        if (!matcher.find()) {
            throw new IllegalStateException(ORDER_NUMBER_NOT_FOUND_ERROR);
        }

        return matcher.group(ORDER_NUMBER_MATCHER_GROUP);
    }
}

File: system-test/typescript/src/testkit/driver/adapter/myShop/ui/client/pages/NewOrderPage.ts
import { BasePage, PAGE_TIMEOUT_MS } from './BasePage.js';

export class NewOrderPage extends BasePage {
  async inputSku(sku: string): Promise<void> {
    await this.page.locator('[aria-label="Product SKU"]').fill(sku, { timeout: PAGE_TIMEOUT_MS });
  }

  async inputQuantity(quantity: string): Promise<void> {
    await this.page.locator('[aria-label="Quantity"]').fill(quantity, { timeout: PAGE_TIMEOUT_MS });
  }

  async inputCountry(country: string): Promise<void> {
    await this.page.locator('[aria-label="Country"]').fill(country, { timeout: PAGE_TIMEOUT_MS });
  }

  async inputCouponCode(couponCode: string): Promise<void> {
    await this.page.locator('[aria-label="Coupon Code"]').fill(couponCode, { timeout: PAGE_TIMEOUT_MS });
  }

  async clickPlaceOrder(): Promise<void> {
    await this.page.locator('[aria-label="Place Order"]').click({ timeout: PAGE_TIMEOUT_MS });
  }

  static getOrderNumber(successMessage: string): string | null {
    const match = /Order has been created with Order Number ([\w-]+)/.exec(successMessage);
    return match ? match[1] : null;
  }
}
